src/game/logic/object/update/swayclientupdate.cpp

```diff
@@ -0,0 +1,138 @@
+/**
+ * @file
+ *
+ * @author xezon
+ *
+ * @brief Sway client update class. Used for foliage movements.
+ *
+ * @copyright Thyme is free software: you can redistribute it and/or
+ *            modify it under the terms of the GNU General Public License
+ *            as published by the Free Software Foundation, either version
+ *            2 of the License, or (at your option) any later version.
+ *            A full copy of the GNU General Public License can be found in
+ *            LICENSE
+ */
+#include "swayclientupdate.h"
+#include "drawable.h"
+#include "matrix3d.h"
+#include "object.h"
+#include "randomvalue.h"
+#include "scriptengine.h"
+
+NameKeyType SwayClientUpdate::s_swayClientUpdateKey = NAMEKEY_INVALID;
+
+SwayClientUpdate::SwayClientUpdate(Thing *thing, const ModuleData *module_data) :
+    ClientUpdateModule(thing, module_data),
+    m_curValue(0.0f),
+    m_curAngle(0.0f),
+    m_curDelta(0.0f),
+    m_curAngleLimit(0.0f),
+    m_leanAngle(0.0f),
+    m_curVersion(-1),
+    m_swaying(true),
+    m_unused(false)
+{
+}
+
+SwayClientUpdate *SwayClientUpdate::Friend_New_Module_Instance(Thing *thing, const ModuleData *module_data)
+{
+    return new SwayClientUpdate(thing, module_data);
+}
+
+void SwayClientUpdate::CRC_Snapshot(Xfer *xfer)
+{
+    DrawableModule::CRC_Snapshot(xfer);
+}
+
+void SwayClientUpdate::Xfer_Snapshot(Xfer *xfer)
+{
+    uint8_t version = 1;
+    xfer->xferVersion(&version, 1);
+    DrawableModule::Xfer_Snapshot(xfer);
+    xfer->xferReal(&m_curValue);
+    xfer->xferReal(&m_curAngle);
+    xfer->xferReal(&m_curDelta);
+    xfer->xferReal(&m_curAngleLimit);
+    xfer->xferReal(&m_leanAngle);
+    xfer->xferShort(&m_curVersion);
+    xfer->xferBool(&m_swaying);
+}
+
+void SwayClientUpdate::Load_Post_Process()
+{
+    ClientUpdateModule::Load_Post_Process();
+    Update_Sway();
+}
+
+NameKeyType SwayClientUpdate::Get_Module_Name_Key() const
+{
+    if (s_swayClientUpdateKey == NAMEKEY_INVALID) {
```

Comment:
In original it tests a static bool here, but I thought we can simplify this and simply check for NAMEKEY_INVALID.

Reply:
its compiler generated on statics
https://github.com/TheAssemblyArmada/Thyme/blob/develop/src/game/logic/object/update/laserupdate.cpp#L78

Reply:
Fixed.